addons/sale/wizard/sale_order_discount.py

```diff
@@ -116,15 +116,14 @@ def _create_discount_lines(self):
                 # No taxes, or all lines have the exact same taxes
                 taxes = next(iter(total_price_per_tax_groups.keys()))
                 subtotal = total_price_per_tax_groups[taxes]
+                dp_name = discount_product.with_context(lang=self.sale_order_id._get_lang()).name
```

Comment:
```suggestion
                partner_lang = self.sale_order_id._get_lang()
                discount_name = discount_product.with_context(lang=partner_lang).name
```
When I suggested `dp_name`, I thought the `dp` in `discount_dp` stood for "discount product", but I just realized it stand for decimal precision, so we should probably use a different name